Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -0,0 +1,598 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.IO;
+using Microsoft.Boogie;
+
+
+namespace Microsoft.Dafny {
+  public class PythonCompiler : Compiler {
+    public PythonCompiler(ErrorReporter reporter) : base(reporter) {
+    }
+
+    public override string TargetLanguage => "Python";
+
+    const string DafnySetClass = "_dafny.Set";
+    const string DafnyMultiSetClass = "_dafny.MultiSet";
+    const string DafnySeqClass = "_dafny.Seq";
+    const string DafnyMapClass = "_dafny.Map";
+
+    public override void EmitCallToMain(Method mainMethod, string baseName, ConcreteSyntaxTree wr) {
+      Coverage.EmitSetup(wr);
+      wr.WriteLine("dafny.Helpers.withHaltHandling({0}.{1})", mainMethod.EnclosingClass, mainMethod);
+    }
+
+    protected override ConcreteSyntaxTree CreateStaticMain(IClassWriter cw) {
+      var wr = (cw as PythonCompiler.ClassWriter).MethodWriter;
+      return wr.WriteLine("def Main():");
+    }
+
+    protected override ConcreteSyntaxTree CreateModule(string moduleName, bool isDefault, bool isExtern,
+      string libraryName,
+      ConcreteSyntaxTree wr) {
+      return wr;
+    }
+
+    protected override string GetHelperModuleName() {
+      throw new NotImplementedException();
+    }
+
+    protected override IClassWriter CreateClass(string moduleName, string name, bool isExtern, string fullPrintName,
+      List<TypeParameter> typeParameters,
+      TopLevelDecl cls, List<Type> superClasses, IToken tok, ConcreteSyntaxTree wr) {
+
+      var w = wr.WriteLine("class {0}:", name);
+
+      var methodWriter = w.NewBlock(open: BraceStyle.Pindent, close: BraceStyle.Pindent);
```

Comment:
You might want to abstract this call to NewBlock into a method if you are going to use it in many places.

Reply:
Not doing it for now, but surely can do as number of call increases.